Feature/kas 2556 link existing publication

erikap opened the pull request:
N.a.v. enkele bemerkingen van @mdv-professional nadat de vorige KAS-2556 branch gemerged is:
- verzekeren dat `case` ingeladen is alvorens de lijst van documenten in de modal getoond wordt
- zorgen dat de link status teruggezet wordt naar 'Geen' indien de modal om een nieuwe publicatie aan te maken geannuleerd wordt

Voor de volledigheid, extra context uit chats over het afsplitsen van de `batch-documents-publication-row`-component.
Mijn voornaamste argumenten voor een aparte component:
* het hoofddoel van de modal is zorgen dat de lijst van document getoond wordt in een model. Het exacte gedrag en mogelijke interacties van iedere rij (vb. linken van bestaande publicatie, openen van document view, ...) horen niet tot dat hoofddoel m.i.
* het gebruik van de `Documents::DocumentList` in de template om de documenten correct te groeperen per container. Op die manier wordt de groeperingslogica niet gedupliceerd in de modal component
* zonder aparte component moet de state van iedere rij gemanaged worden in de modal component, wat die component veel complexer maakt dan nodig, voornl. omdat die component op een andere grootteorde werkt (lijst van documenten vs 1 document per rij)
* het gebruik van de `Documents::DocumentList` is in lijn met de andere plaatsen in de code waar dit gebruikt wordt. De component geeft je een lijst van laatste document versies. Door een block mee te geven bepaal je hoe deze gerenderd worden. Andere mogelijke renders die er nu zijn, zijn de `Documents::DocumentBadge` en de `Documents::DocumentCard`. In feite is de `Publication::BatchDocumentsPublicationRow` een extra soort rendering die we toevoegen.

erikap committed:
Await loading of case before rendering list of documents for publication

erikap committed:
Move new publication modal management to batch-documents-publication-row

Commit:
Merge branch 'feature/KAS-2556-link-existing-publication' of https://github.com/kanselarij-vlaanderen/kaleidos-frontend into KAS-2556-pub-from-mr--link-doc-to-pub

mdv-professional commented on app/components/publications/batch-documents-publication-modal.hbs:
```diff
@@ -1,7 +1,7 @@
   <Auk::Modal::Header @title={{ t "documents" }} @onClose={{@onClose}} />
   <Auk::Modal::Body>
-    <Utils::LoadableContent @isLoading={{this.loadPieces.isRunning}}>
+    <Utils::LoadableContent @isLoading={{or this.loadPieces.isRunning this.loadCase.isRunning}}>
```
Akkoord

mdv-professional commented on app/components/publications/batch-documents-publication-row.hbs:
```diff
@@ -49,3 +49,10 @@
+    @case={{this.case}}
+    @onSave={{this.saveNewPublication}}
+    @onCancel={{this.cancelNewPublication}} />
+{{/if}}
```
### NewPublicationModal
Niet akkoord: Dit zorgt ervoor dat de NewPublicationModal getoond wordt binnen de css van de met een css-class als full-screen gemarkeerde BatchDocumentsPublicationModal. Hierdoor is de NewPublicationModal ook fullscreen. Ik heb er ook moeite mee dat er dan (code-matig) een NewPublicationModal bestaat voor elke row, maar dat kan aan mij liggen.

Ik zie niet meteen een properdere workaround. Daarom zou ik best werken met een context object. Als ge de document-grouping wilt centraal houden zou ik hiervoor een static functie toevoegen aan Piece.

Een andere mogelijkheid misschien: Ik heb gezien dat de powerselect de Ember-Wormhole-addon gebruikt.

Ik vind het ook jammer dat de fade-out animatie bij sluiten van de modal niet in de applicatie zit.

### selected property
Met de selected computed property zorgde ik ervoor dat de juiste publicatieflow in de optionslijst van de selectbox geselecteerd was. Dat is iets vreemd met EmberPowerSelect: het model meegegeven met @onchange(selected) en het model in de lijst van @options zijn in sommige gevallen niet dezelfde instanties. Ik ben er nog niet uit wanneer dit zo is, maar in deze modal was dit zo.

erikap replied:
Over de positionering van de new publication modal heb je gelijk. Ik heb deze terug verhuisd naar de batch-documents-publication-modal. Met een `did-update` modifier wordt er nu in de row-component gezorgd dat de selected option aangepast wordt wanneer de publication flow van het document wijzigt.

mdv-professional reviewed (changes requested):
## i.v.m. de selected property
Met de selected computed property zorgde ik ervoor dat de juiste publicatieflow in de optionslijst van de selectbox geselecteerd was. Dat is iets vreemd met EmberPowerSelect: het model meegegeven met @onchange(selected) en het model in de lijst van @options zijn in sommige gevallen niet dezelfde instanties. Ik ben er nog niet uit wanneer dit zo is, maar in deze modal was dit zo.

Commit:
KAS-2556 @case | link doc to pub [pub from mr]

erikap committed:
Merge branch 'development' into feature/KAS-2556-link-existing-publication

erikap committed:
Move new publication modal back to batch-documents-publication-modal

Using did-update render modifier to keep selected option of
batch-documents-publication-row up-to-date. Should probably be changed
to an `@use` resource once we've upgraded to an Ember version
supporting the https://github.com/pzuraq/ember-could-get-used-to-this addon

erikap committed:
Fix selected option in publication-flow-selector component

power-select only correctly highlights the current selected value if a
non-proxy object is passed. Hence, we need to resolve the relation
promise before passing it as an argument to power-select

erikap committed:
Merge branch 'feature/KAS-2556-link-existing-publication' of github.com:kanselarij-vlaanderen/kaleidos-frontend into feature/KAS-2556-link-existing-publication

erikap committed:
Merge branch 'development' into feature/KAS-2556-link-existing-publication

erikap commented:
> **i.v.m. de selected property**
> Met de selected computed property zorgde ik ervoor dat de juiste publicatieflow in de optionslijst van de selectbox geselecteerd was. Dat is iets vreemd met EmberPowerSelect: het model meegegeven met @onchange(selected) en het model in de lijst van @options zijn in sommige gevallen niet dezelfde instanties. Ik ben er nog niet uit wanneer dit zo is, maar in deze modal was dit zo.

Ik begrijp hier nu wat je bedoelt: de geselecteerde waarde werd niet correct gehighlight in de lijst van opties. EmberPowerSelect aanziet een proxy-object (vb `@piece.publicationFlow`) blijkbaar niet als hetzelfde als de resolved value. Ik heb dit aangepast zodat de resolved value doorgegeven wordt  als `@selected` aan EmberPowerSelect. De geselecteerde optie wordt nu wel correct gehighlight.

mdv-professional reviewed (approved):
Het werkt goed, en ik begrijp het ook, maar ik vind de did-update-weg niet ideaal. Ik denk nog steeds dat een table en een row (hier) te veel aan elkaar gelinkt zijn om te scheiden. Modifiers zijn in mijn ogen niet de meest straight-forward manier om dit te schrijven, maar dat is mijn aanvoelen.

ghost commented:
Nog een opmerking: de nieuwe modal component heeft standaard een margin. Hier is dit niet gewenst.

UPDATE: aangepast

Commit:
KAS-2556 no padding in modal

Commit:
Merge branch 'development' into feature/KAS-2556-link-existing-publication
